plugins/Amplifier/Amplifier.cpp

```diff
@@ -65,17 +67,17 @@ bool AmplifierEffect::processAudioBuffer(sampleFrame* buf, const fpp_t frames)
 	const float d = dryLevel();
 	const float w = wetLevel();
 
-	const ValueBuffer* volumeBuf = m_ampControls.m_volumeModel.valueBuffer();
-	const ValueBuffer* panBuf = m_ampControls.m_panModel.valueBuffer();
-	const ValueBuffer* leftBuf = m_ampControls.m_leftModel.valueBuffer();
-	const ValueBuffer* rightBuf = m_ampControls.m_rightModel.valueBuffer();
+	const auto* volumeBuf = m_ampControls.m_volumeModel.valueBuffer();
+	const auto* panBuf = m_ampControls.m_panModel.valueBuffer();
+	const auto* leftBuf = m_ampControls.m_leftModel.valueBuffer();
+	const auto* rightBuf = m_ampControls.m_rightModel.valueBuffer();
 
-	for (fpp_t f = 0; f < frames; ++f)
+	for (size_t f = 0; f < frames; ++f)
 	{
-		const float volume = (volumeBuf ? volumeBuf->value(f) : m_ampControls.m_volumeModel.value()) * 0.01f;
-		const float pan = (panBuf ? panBuf->value(f) : m_ampControls.m_panModel.value()) * 0.01f;
-		const float left = (leftBuf ? leftBuf->value(f) : m_ampControls.m_leftModel.value()) * 0.01f;
-		const float right = (rightBuf ? rightBuf->value(f) : m_ampControls.m_rightModel.value()) * 0.01f;
+		const float volume = (volumeBuf ? volumeBuf->at(std::clamp(f,(size_t) 0, volumeBuf->size())) : m_ampControls.m_volumeModel.value()) * 0.01f;
```

Comment:
You can also explicitly specify the template type, e.g., `std::clamp<std::size_t>(f, 0, volumeBuf->size())`.